### PR TITLE
Add EFI netboot support

### PR DESCRIPTION
**Pull request by mjg59**

### CODE REVIEW

**Comment by mjg59** [comment]
Depends on https://github.com/coreos/grub/pull/21

**Comment by mjg59** [comment]
Also depends on https://github.com/coreos/grub/pull/22

File: build_library/grub.cfg [hunk 1]
```diff
@@ -27,9 +27,31 @@ if [ "$grub_platform" = "efi" ]; then
...
 
+if [ "$net_default_server" != "" ]; then
+   smbios --type 1 --get-uuid 8 --set uuid
+   smbios --type 1 --get-string 7 --set serial
+   set mac=$net_default_mac
+   # Re-DHCP as grub to get the API endpoint
+   net_bootp $net_default_interface
+   # This is awful, but grub doesn't support a nice way to do this
+   for i in "$net_efinet0_dhcp_boot_file" "$net_efinet1_dhcp_boot_file" "$net_efinet2_dhcp_boot_file" "$net_efinet3_dhcp_boot_file" "$net_efinet4_dhcp_boot_file" "$net_efinet5_dhcp_boot_file" "$net_efinet6_dhcp_boot_file" "$net_efinet7_dhcp_boot_file" "$net_efinet8_dhcp_boot_file" "$net_efinet9_dhcp_boot_file"; do
+      if [ $i != "" ]; then
```
Comment (marineam) [inline build_library/grub.cfg]: Probably should quote `$i` here and below.

File: build_library/grub.cfg [hunk 2]
```diff
@@ -27,9 +27,31 @@ if [ "$grub_platform" = "efi" ]; then
...
+          set endpoint=$i
+          break
+      fi
+   done
```
Comment (marineam) [inline build_library/grub.cfg]: Perhaps check for endpoint being empty

File: build_library/grub.cfg [hunk 3]
```diff
@@ -27,9 +27,31 @@ if [ "$grub_platform" = "efi" ]; then
    getenv -e SetupMode -g 8be4df61-93ca-11d2-aa0d-00e098032b8c -b setupmode
    if [ "$sb" = "01" -a "$setupmode" = "00" ]; then
       set secure_boot="1"
+      getenv -e gpg -g b8ade7d5-d400-4213-8d15-d47be0a621bf -b gpg
```
Comment (marineam) [inline build_library/grub.cfg]: Seems like the name in UEFI should be more descriptive.

Also should check that a value was found before trusting it.

**Comment by mjg59** [comment]
Updated based on comments

File: build_library/grub.cfg [hunk 4]
```diff
@@ -27,6 +27,32 @@ if [ "$grub_platform" = "efi" ]; then
    getenv -e SetupMode -g 8be4df61-93ca-11d2-aa0d-00e098032b8c -b setupmode
    if [ "$sb" = "01" -a "$setupmode" = "00" ]; then
       set secure_boot="1"
+      getenv -e gpgpubkey -g b8ade7d5-d400-4213-8d15-d47be0a621bf -b gpgpubkey
```
Comment (marineam) [inline build_library/grub.cfg]: I meant descriptive name for what it is used for, GrubVerificationKey, or NetworkBootVerificationKey or something.

**Comment by mjg59** [comment]
Got it. This should be better.

**Comment by marineam** [comment]
LGTM

**Pull request merged**